src/generateIncludes.js

```diff
@@ -63,67 +61,70 @@ export default function generateIncludes(simpleAST, type, context, options) {
       includeOptions = argsToFindOptions(args, association.target);
       allowedAttributes = Object.keys(association.target.rawAttributes);
 
-
       if (options.filterAttributes) {
         includeOptions.attributes = (includeOptions.attributes || [])
-                                  .concat(Object.keys(fieldAST.fields).map(key => fieldAST.fields[key].key || key))
-                                  .filter(inList.bind(null, allowedAttributes));
+          .concat(Object.keys(fieldAST.fields).map(key => fieldAST.fields[key].key || key))
+          .filter(inList.bind(null, allowedAttributes));
       } else {
         includeOptions.attributes = allowedAttributes;
       }
 
-      if (includeResolver.$before) {
-        includeOptions = includeResolver.$before(includeOptions, args, context, {
-          ast: fieldAST,
-          type: type
-        });
-      }
+      return new Promise(function (resolve) {
+        if (includeResolver.$before) {
+          resolve(includeResolver.$before(includeOptions, args, context, {
+            ast: fieldAST,
+            type: type
+          }));
+        } else {
+          resolve(includeOptions);
+        }
```

Comment:
Why new promise? If it's because you need a chain `Promise.resolve().then(` might be better.